client/gutenberg/extensions/publicize/publicize-settings-button.jsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Publicize settings button component.
+ *
+ * Component which allows user to click to open settings
+ * in a new window/tab. If window/tab is closed, then
+ * connections will be automatically refreshed.
+ *
+ * @since  5.9.1
+ */
+
+// Since this is a Jetpack originated block in Calypso codebase,
+// we're relaxing className and some accessibility rules.
+/* eslint wpcalypso/jsx-classname-namespace: 0 */
+/* eslint jsx-a11y/anchor-is-valid: 0 */
+/* eslint jsx-a11y/click-events-have-key-events: 0 */
+/* eslint jsx-a11y/no-static-element-interactions: 0 */
+/* eslint jsx-a11y/no-noninteractive-tabindex: 0 */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+const { __ } = window.wp.i18n;
+
+class PublicizeSettingsButton extends Component {
+	/**
+	 * Opens up popup so user can view/modify connections
+	 *
+	 * @since 5.9.1
+	 *
+	 * @param {object} event Event instance for onClick.
+	 */
+	settingsClick = ( event ) => {
+		const href = 'options-general.php?page=sharing&publicize_popup=true';
+		const { refreshCallback } = this.props;
+		event.preventDefault();
+		/**
+		 * Open a popup window, and
+		 * when it is closed, refresh connections
+		 */
+		const popupWin = window.open( href, '', '' );
+		const popupTimer = window.setInterval( () => {
+			if ( false !== popupWin.closed ) {
+				window.clearInterval( popupTimer );
+				refreshCallback();
+			}
+		}, 500 );
```

Comment:
it would be favorable not to use the `setInterval()` here - I'm curious why it's here. in general I think it's nice to rely on active propagation of the event rather than passive as is done here. that is, we should make a decision to check again in the future instead of relying on the system to automatically keep it going (that way we can prevent accidental background tasks from forming)

also, why was 500 chosen as the timeout value? (I'm starting to wonder if this was copy/pasted from Stack-Overflow without citation). There's a very similar solution there that uses `1000`